src/main/java/app/nzyme/core/rest/resources/dot11/BanditsResource.java

```diff
@@ -0,0 +1,277 @@
+            if (b.id().equals(id)) {
+                bandit = b;
+                break;
+            }
+        }
+
+        if (bandit == null) {
+            return Response.status(Response.Status.NOT_FOUND).build();
+        }
+
+        return Response.ok(BuiltinBanditDetailsResponse.create(
+                bandit.id(),
+                bandit.name(),
+                bandit.description(),
+                bandit.fingerprints() == null ? Collections.emptyList() : bandit.fingerprints()
+        )).build();
+    }
+
+    @GET
+    @RESTSecured(value = PermissionLevel.ANY, featurePermissions = { "dot11_monitoring_manage" })
+    @Path("/custom")
+    public Response findAllCustom(@Context SecurityContext sc,
+                                  @QueryParam("limit") int limit,
+                                  @QueryParam("offset") int offset,
+                                  @QueryParam("organization_uuid") @NotNull UUID organizationId,
+                                  @QueryParam("tenant_uuid") @NotNull UUID tenantId) {
+        AuthenticatedUser authenticatedUser = getAuthenticatedUser(sc);
+
+        if (limit > 250) {
+            LOG.warn("Requested limit larger than 250. Not allowed.");
+            return Response.status(Response.Status.UNAUTHORIZED).build();
+        }
+
+        if (!hasPermissions(authenticatedUser, organizationId, tenantId)) {
+            return Response.status(Response.Status.NOT_FOUND).build();
+        }
+
+        long total = nzyme.getDot11().countCustomBandits(organizationId, tenantId);
+        List<CustomBanditDetailsResponse> bandits = Lists.newArrayList();
+
+        for (CustomBanditDescription b : nzyme.getDot11().findAllCustomBandits(organizationId, tenantId, limit, offset)) {
+            // Find fingerprints of bandit.
+            List<String> fingerprints = nzyme.getDot11().findFingerprintsOfCustomBandit(b.id());
+
+            bandits.add(CustomBanditDetailsResponse.create(
+                    b.uuid(),
+                    b.name(),
+                    b.description(),
+                    fingerprints,
+                    b.createdAt(),
+                    b.updatedAt()
+            ));
+        }
+
+        return Response.ok(CustomBanditListResponse.create(total, bandits)).build();
+    }
+
+    @GET
+    @RESTSecured(value = PermissionLevel.ANY, featurePermissions = { "dot11_monitoring_manage" })
+    @Path("/custom/show/{id}")
+    public Response findCustom(@Context SecurityContext sc,
+                               @PathParam("id") @NotNull UUID id) {
+        AuthenticatedUser authenticatedUser = getAuthenticatedUser(sc);
+
+        Optional<CustomBanditDescription> bandit = nzyme.getDot11().findCustomBandit(id);
+
+        if (bandit.isEmpty()
+                || !hasPermissions(authenticatedUser, bandit.get().organizationId(), bandit.get().tenantId())) {
+            return Response.status(Response.Status.NOT_FOUND).build();
+        }
```

Comment:
Maybe you are returning a NOT_FOUND for a permission denied request because how you engineered the Web UI, but in case it is an easy change, same recommendation. UNAUTHORIZED for the permission denied and NOT_FOUND for the empty case.